python/ray/rllib/agents/dqn/dqn.py

```diff
@@ -70,7 +71,7 @@
     "exploration_final_eps": 0.02,
     # Update the target network every `target_network_update_freq` steps.
     "target_network_update_freq": 500,
-    # Use softmax for sampling actions.
+    # Use softmax for sampling actions. Required for off policy estimation.
```

Comment:
Why is soft q required for off policy estimation?

Reply:
Otherwise there's not really a meaningful probability distribution over actions (just eps-greedy exploration). Horizon does the same thing with requiring soft-Q for OPE.